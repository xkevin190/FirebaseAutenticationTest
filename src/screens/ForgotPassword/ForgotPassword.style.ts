import {StyleSheet, ViewStyle} from 'react-native';

import {COLORS} from '../../constants/styles';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    flexDirection: 'column',
    backgroundColor: COLORS.GRAY_LIGHT,
  } as ViewStyle,
  inputsContainer: {
    width: '100%',
    paddingHorizontal: 24,
    marginBottom: 8,
    marginTop: 40,
  } as ViewStyle,
  sendEmailButtonContainer: {
    width: '100%',
    paddingHorizontal: 24,
  } as ViewStyle,
});

export default styles;
