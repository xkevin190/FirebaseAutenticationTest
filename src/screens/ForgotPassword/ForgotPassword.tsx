import React, {useState, useCallback} from 'react';
import {
  View,
  TouchableWithoutFeedback,
  Keyboard,
  KeyboardAvoidingView,
} from 'react-native';
import {useTranslation} from 'react-i18next';
import {CustomInput, Button} from '../../shared';
import {COLORS} from '../../constants/styles';
import ForgotPasswordCompleted from './components/ForgotPasswordCompleted';
import styles from './ForgotPassword.style';
import Controller from '../../shared/Controller';
import {useForgotPasswordForm} from './hooks/useForgotPasswordForm';

const ControlledInputs = Controller(CustomInput);

const ForgotPassword: React.FC = () => {
  const loading = false; //useAppSelector(getRecoverPasswordLoading);
  const sucessfull = false; //
  const [email, setEmail] = useState('');
  const {t} = useTranslation();

  const {control, watch, handleSubmit} = useForgotPasswordForm({
    email: '',
  });

  const onSubmit = (data: {email: string}) => {};

  return sucessfull ? (
    <ForgotPasswordCompleted />
  ) : (
    <TouchableWithoutFeedback onPress={Keyboard.dismiss}>
      <View style={styles.container}>
        <KeyboardAvoidingView style={styles.inputsContainer} behavior="padding">
          <ControlledInputs
            name="email"
            control={control}
            value={watch('email')}
            title={t('forgotPassword:translation.emailInput.label')}
            placeholder={t('forgotPassword:translation.emailInput.placeholder')}
          />
        </KeyboardAvoidingView>
        <View style={styles.sendEmailButtonContainer}>
          <Button
            buttonText={t(
              `forgotPassword:translation.sendEmailButton.${
                loading ? 'loading' : 'label'
              }`,
            )}
            textColor={COLORS.WHITE}
            rounded
            size="xxl"
            onPress={handleSubmit(onSubmit)}
            disabled={loading}
          />
        </View>
      </View>
    </TouchableWithoutFeedback>
  );
};

export default ForgotPassword;
