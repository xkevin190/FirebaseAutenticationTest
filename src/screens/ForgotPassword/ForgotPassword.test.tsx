import React from 'react';
import {
  render,
  fireEvent,
  RenderResult,
  waitFor,
} from '@testing-library/react-native';
import {NavigationContext} from '@react-navigation/native';
import ForgotPassword from './ForgotPassword';

let mockDispatch = jest.fn();
jest.mock('react-native-vector-icons/Ionicons', () => 'Icon');

jest.mock('../../hooks/useAppDispatch',  () => {
    return ()=> mockDispatch;
});

jest.mock('../../hooks/useAppSelector', () => jest.fn());
jest.mock('../../store/Auth/selectors', () => jest.fn());

jest.mock('react-i18next', () => ({
  // this mock makes sure any components using the translate hook can use it without a warning being shown
  useTranslation: () => {
    return {
      t: (str: string) => str,
      i18n: {
        changeLanguage: () => new Promise(() => {}),
      },
    };
  },
}));

jest.mock('../../store/Auth/slice', () => {
  return {
    resetPasswordThunk: jest.fn(),
    authActions: {
      resetError: jest.fn(),
      resetPasswordSent: jest.fn(),
    },
  };
});

const navContext = {
    isFocused: () => true,
    addListener: jest.fn(() => jest.fn()),
  };

let Component: RenderResult;

beforeEach(() => {
  Component = render(
    <NavigationContext.Provider value={navContext}>
      <ForgotPassword />
    </NavigationContext.Provider>,
  );
});


describe('ForgotPassword component',  () => {
  test('renders all input fields',  () => {
    const {getByPlaceholderText, toJSON} = Component;
   
    const emailInput = getByPlaceholderText(
      'forgotPassword:translation.emailInput.placeholder',
    );

    expect(toJSON()).toMatchSnapshot();
    expect(emailInput).toBeTruthy();
  });

  test('submits the form with valid email', async () => {
    const {getByPlaceholderText, getByTestId} = Component;

    const emailInput = getByPlaceholderText(
      'forgotPassword:translation.emailInput.placeholder',
    );
  
    const submitButton = getByTestId('resetPasswordButton');
    fireEvent.changeText(emailInput, 'test@gmail.com');
    expect(emailInput.props.value).toBe('test@gmail.com');

    await waitFor(() => {
        fireEvent.press(submitButton) 
    });
    expect(mockDispatch).toHaveBeenCalled();
    
  });

  test('displays error message for empty email field', async () => {
    const {getByPlaceholderText,getByTestId, queryAllByTestId} =
      Component;
    
    const emailInput = getByPlaceholderText(
      'forgotPassword:translation.emailInput.placeholder',
    );
   
    const submitButton = getByTestId('resetPasswordButton');

    fireEvent.changeText(emailInput, '');
    fireEvent.press(submitButton);

    await waitFor(() => {
      expect(queryAllByTestId('helperText')).toHaveLength(1);
    });
    
  });
});
