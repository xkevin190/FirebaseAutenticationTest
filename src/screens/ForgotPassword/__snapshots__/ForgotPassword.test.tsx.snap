// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForgotPassword component renders all input fields 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#F6F8FF",
      "flex": 1,
      "flexDirection": "column",
    }
  }
>
  <View
    onLayout={[Function]}
    style={
      [
        {
          "marginBottom": 8,
          "marginTop": 40,
          "paddingHorizontal": 24,
          "width": "100%",
        },
        {
          "paddingBottom": 0,
        },
      ]
    }
  >
    <Text
      style={
        {
          "color": "#9DA5B9",
          "fontFamily": "Lato-Bold",
          "fontSize": 12,
        }
      }
      testID=""
    >
      FORGOTPASSWORD:TRANSLATION.EMAILINPUT.LABEL
    </Text>
    <View
      style={
        {
          "backgroundColor": "#FFFFFF",
          "borderColor": "#9DA5B9",
          "borderRadius": 16,
          "borderWidth": 1,
          "flexDirection": "row",
          "marginBottom": 18,
          "marginTop": 6,
          "width": "100%",
        }
      }
    >
      <TextInput
        autoCapitalize="none"
        autoCorrect={false}
        control={
          {
            "_defaultValues": {
              "email": "",
            },
            "_disableForm": [Function],
            "_executeSchema": [Function],
            "_fields": {
              "email": {
                "_f": {
                  "mount": true,
                  "name": "email",
                  "ref": {
                    "name": "email",
                  },
                  "value": "",
                },
              },
            },
            "_formState": {
              "dirtyFields": {},
              "disabled": false,
              "errors": {},
              "isDirty": false,
              "isLoading": false,
              "isSubmitSuccessful": false,
              "isSubmitted": false,
              "isSubmitting": false,
              "isValid": false,
              "isValidating": false,
              "submitCount": 0,
              "touchedFields": {},
              "validatingFields": {},
            },
            "_formValues": {
              "email": "",
            },
            "_getDirty": [Function],
            "_getFieldArray": [Function],
            "_getWatch": [Function],
            "_names": {
              "array": Set {},
              "mount": Set {
                "email",
              },
              "unMount": Set {},
              "watch": Set {
                "email",
              },
            },
            "_options": {
              "defaultValues": {
                "email": "",
              },
              "delayError": 100,
              "mode": "onBlur",
              "reValidateMode": "onChange",
              "resolver": [Function],
              "shouldFocusError": true,
            },
            "_proxyFormState": {
              "dirtyFields": false,
              "disabled": true,
              "errors": true,
              "isDirty": false,
              "isValid": false,
              "isValidating": false,
              "touchedFields": false,
              "validatingFields": false,
            },
            "_removeUnmounted": [Function],
            "_reset": [Function],
            "_resetDefaultValues": [Function],
            "_setErrors": [Function],
            "_state": {
              "action": false,
              "mount": true,
              "watch": false,
            },
            "_subjects": {
              "array": {
                "next": [Function],
                "observers": [],
                "subscribe": [Function],
                "unsubscribe": [Function],
              },
              "state": {
                "next": [Function],
                "observers": [
                  {
                    "next": [Function],
                  },
                  {
                    "next": [Function],
                  },
                ],
                "subscribe": [Function],
                "unsubscribe": [Function],
              },
              "values": {
                "next": [Function],
                "observers": [
                  {
                    "next": [Function],
                  },
                ],
                "subscribe": [Function],
                "unsubscribe": [Function],
              },
            },
            "_updateDisabledField": [Function],
            "_updateFieldArray": [Function],
            "_updateFormState": [Function],
            "_updateValid": [Function],
            "getFieldState": [Function],
            "handleSubmit": [Function],
            "register": [Function],
            "setError": [Function],
            "unregister": [Function],
          }
        }
        multiline={false}
        name="email"
        numberOfLines={1}
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        placeholder="forgotPassword:translation.emailInput.placeholder"
        placeholderTextColor="#9DA5B9"
        secureTextEntry={false}
        style={
          {
            "color": "#000000",
            "flex": 1,
            "fontFamily": "Lato-Regular",
            "fontSize": 14,
            "paddingHorizontal": 16,
            "paddingVertical": 10,
          }
        }
        textAlignVertical="auto"
        value=""
      />
    </View>
  </View>
  <View
    style={
      {
        "paddingHorizontal": 24,
        "width": "100%",
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": false,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#2162E3",
          "borderRadius": 25,
          "height": 50,
          "justifyContent": "center",
          "opacity": 1,
          "width": "100%",
        }
      }
      testID="resetPasswordButton"
    >
      <Text
        style={
          {
            "color": "#FFFFFF",
            "fontFamily": "Lato-Regular",
            "fontSize": 14,
          }
        }
        testID=""
      >
        forgotPassword:translation.sendEmailButton.label
      </Text>
    </View>
  </View>
</View>
`;
